src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicSession.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Security;
+using System.Text;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    internal sealed class MsQuicSession : IDisposable
+    {
+        private bool _disposed = false;
+        private IntPtr _nativeObjPtr;
+        private MsQuicApi _api;
+        private bool _opened;
+
+        internal MsQuicSession(string registration)
+        {
+            _api = new MsQuicApi();
+
+            _api.RegistrationOpen(Encoding.ASCII.GetBytes(registration));
```

Comment:
Is the ASCII encoding the right one to use here? I see other interop definitions are using `UnmanagedType.LPStr` (ie current Windows code page) or UTF8.

Reply:
This field is purely for debugging attribution. It's never sent over the wire, so encoding doesn't really matter, but if I had to make a suggestion, it'd be UTF8.